samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/translation/TranslationMiddleware.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.multilingual.translation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Strings;
+import com.microsoft.bot.builder.Middleware;
+import com.microsoft.bot.builder.NextDelegate;
+import com.microsoft.bot.builder.StatePropertyAccessor;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.builder.UserState;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ActivityTypes;
+
+/**
+ * Middleware for translating text between the user and bot.
+ * Uses the Microsoft Translator Text API.
+ */
+public class TranslationMiddleware implements Middleware  {
+    private MicrosoftTranslator translator;
+    private StatePropertyAccessor<String> languageStateProperty;
+
+    /**
+     * Initializes a new instance of the {@link TranslationMiddleware} class.
+     * @param withTranslator Translator implementation to be used for text translation.
+     * @param userState State property for current language.
+     */
+    public TranslationMiddleware(MicrosoftTranslator withTranslator, UserState userState) {
+        if (withTranslator == null) {
+            throw new IllegalArgumentException("withTranslator");
+        }
+        this.translator = withTranslator;
+        if (userState == null) {
+            throw new IllegalArgumentException("userState");
+        }
+
+        this.languageStateProperty = userState.createProperty("LanguagePreference");
+    }
+
+    /**
+     * Processes an incoming activity.
+     * @param turnContext Context object containing information for a single turn of conversation with a user.
+     * @param next The delegate to call to continue the bot middleware pipeline.
+     * @return A Task representing the asynchronous operation.
+     */
+    public CompletableFuture<Void> onTurn(TurnContext turnContext, NextDelegate next) {
+        if (turnContext == null) {
+            throw new IllegalArgumentException("turnContext");
+        }
+
+        return this.shouldTranslate(turnContext).thenCompose(translate -> {
+            if (translate) {
+                if (turnContext.getActivity().getType() == ActivityTypes.MESSAGE) {
+                    return this.translator.translate(turnContext.getActivity().getText(), TranslationSettings.DEFAULT_LANGUAGE).thenApply(text -> {
```

Comment:
There are several lines over 120 chars and that fires the checkstyle warning, are we going to follow that rule?